app/controllers/users_controller.rb

```diff
@@ -521,6 +522,36 @@ def admin_login
     render layout: false
   end
 
+  def email_login
+    return redirect_to path("/") if current_user
+
+    expires_now
+    params.require(:login)
+
+    RateLimiter.new(nil, "email-login-hour-#{request.remote_ip}", 6, 1.hour).performed!
+    RateLimiter.new(nil, "email-login-min-#{request.remote_ip}", 3, 1.minute).performed!
+
+    RateLimiter.new(nil, "email-login-hour-#{params[:login].to_s[0..100]}", 12, 1.hour).performed!
+    RateLimiter.new(nil, "email-login-min-#{params[:login].to_s[0..100]}", 3, 1.minute).performed!
+
+    user = User.human_users.find_by_username_or_email(params[:login])
+    user_presence = user.present? && !user.staged
+    if user_presence
+      email_token = user.email_tokens.create(email: user.email)
+      Jobs.enqueue(:critical_user_email, type: :email_login, user_id: user.id, email_token: email_token.token)
+    end
+
+    json = { result: "ok" }
+    unless SiteSetting.forgot_password_strict
+      json[:user_found] = user_presence
+    end
+
+    render json: json
+
+  rescue RateLimiter::LimitExceeded
+    render_json_error(I18n.t("rate_limiter.slow_down"))
```

Comment:
Since we're using the default rate limiter message here. We can just let [application_controller](https://github.com/discourse/discourse/blob/master/app/controllers/application_controller.rb#L101-L103) rescue the error.